qualtran/bloqs/block_encoding/sparse_matrix.py

```diff
@@ -281,6 +284,65 @@ def build_composite_bloq(self, bb: BloqBuilder, **soqs: SoquetT) -> Dict[str, So
         return soqs
 
 
+@frozen
+class SymmetricBandedRowColumnOracle(RowColumnOracle):
+    """Oracle specifying the non-zero rows and columns of a symmetric banded matrix.
+
+    Args:
+        system_bitsize: The number of bits used to represent an index.
+        bandsize: The number of pairs of non-zero off-main diagonals. A diagonal matrix has
+            bandsize 0 and a tridiagonal matrix has bandsize 1.
+
+    Registers:
+        l: As input, index specifying the `l`-th non-zero entry to find in row / column `i`.
+           As output, position of the `l`-th non-zero entry in row / column `i`.
+        i: The row / column index.
+    """
+
+    system_bitsize: SymbolicInt
+    bandsize: SymbolicInt
+
+    @cached_property
+    def num_nonzero(self) -> SymbolicInt:
+        return 2 * self.bandsize + 1
+
+    def __attrs_post_init__(self):
+        if is_symbolic(self.system_bitsize) or is_symbolic(self.bandsize):
+            return
+        if 2**self.system_bitsize < 2 * self.bandsize:
+            raise ValueError(
+                f"bandsize={self.bandsize} too large for system_bitsize={self.system_bitsize}"
+            )
+
+    def call_classically(self, l: ClassicalValT, i: ClassicalValT) -> Tuple[ClassicalValT, ...]:
+        if (
+            is_symbolic(self.bandsize)
+            or is_symbolic(self.system_bitsize)
+            or is_symbolic(self.num_nonzero)
```

Comment:
You're right that it would still work. Is it expected that we would ever `call_classically` a bloq with symbolic attributes but concrete `ClassicalValT`?